rasterio/windows.py

```diff
@@ -195,6 +213,33 @@ def from_bounds(left, bottom, right, top, transform,
         return crop(window, height, width)
 
 
+def int_reshape(window, pixel_precision=3):
+    """Converts floating point value Windows to integer value Windows.
+
+    Parameters
+    ----------
+    window : Window
+        Input window with floating point values.
+    pixel_precision : int
+        Rounding precision in decimal places.
+    Returns
+    -------
+    Window
+        A new Window
+    """
+    if isinstance(window, tuple):
+        warn_window_deprecation()
+        return Window.from_offlen(
+            window[1][0], window[0][0],
+            math.ceil(round(window[1][1] - window[1][0], pixel_precision)),
+            math.ceil(round(window[0][1] - window[0][0], pixel_precision)))
+    else:
+        return Window.from_offlen(
+            window.col_off, window.row_off,
+            math.ceil(round(window.num_cols, pixel_precision)),
+            math.ceil(round(window.num_rows, pixel_precision)))
```

Comment:
Returns a new window with the same offsets, but with integer height and width (as float). The `pixel_precision` parameter helps us snap to .001 pixel precision.